core/src/com/cloud/agent/api/StartAnswer.java

```diff
@@ -24,9 +24,12 @@
 import com.cloud.agent.api.to.VirtualMachineTO;
 
 public class StartAnswer extends Answer {
+    public static final String PATH = "path";
+    public static final String IMAGE_FORMAT = "imageFormat";
+
     VirtualMachineTO vm;
     String hostGuid;
-    Map<String, String> _iqnToPath;
+    private Map<String, Map<String, String>> _iqnToData;
```

Comment:
I could go either way on it. It's just a Map whose key (a String) returns another Map.

Reply:
With regards to multiple PRs: Originally I had it that way (VMware in one PR, KVM in another, and XenServer in a third). Unfortunately, since 4.10 took a very long to to come out, it became a big pain to constantly rebase and maintain all of those PRs. As such, I merged the code into this one to simplify PR maintenance and to minimize the chances of making mistakes with rebasing due to having to perform that action so many times (and having to make so many manual merge fixes along the way). At this point, I think it's safest to stay with this one, large PR.

Feel free to provide your code style and maintainability suggestions. I'm happy to hear what you have to say.

Reply:
What is this `iqnToData`? What is the key of the first map? And the key of the second map?